File: jump-game.cpp
#include <vector>

using namespace std;

class Solution {
public:
  bool canJump(vector<int> &nums) {
    int jump = 0;

    for (int i = 0; i < nums.size() - 1; ++i) {
      jump = max(jump - 1, nums[i]);
      if (jump == 0) {
        return false;
      }
    }

    return true;
  }
};

File: jump-game.py
from typing import List


class Solution:
    def canJump(self, nums: List[int]) -> bool:
        jump = 0

        for i in range(len(nums) - 1):
            jump = max(jump - 1, nums[i])
            if jump == 0:
                return False

        return True
